src/patch_config.rs

```diff
@@ -553,6 +553,7 @@ pub struct PatchConfig
     pub results_string: Option<String>,
     pub artifact_hints: Option<HashMap<String,String>>, // e.g. "Strength":"This item can be found in Ruined Fountain"
     pub artifact_temple_layer_overrides: Option<HashMap<String,bool>>,
+    pub lore_hints: Option<HashMap<String,String>>,
```

Comment:
HashMap equates to `{ ... }` in JSON, which is, by JSON specification, order *independent*. Your implementation of the parsing of `lore_hints` is order *dependent*.

You need to either:

1) Change lore hints to be a `Vec`

or 

2) Parse `lore_hints` by key rather than by iterating. An elegant key could be the asset ID of the STRG that is to be replaced (this would be a generic catch-all for replacing anything, not just lore).

Reply:
>     2. Parse `lore_hints` by key rather than by iterating. An elegant key could be the asset ID of the STRG that is to be replaced (this would be a generic catch-all for replacing anything, not just lore).

I would prefer this if possible. Being able to change any STRG in the game would be useful, and several other STRG-related changes in the patcher could be changed to just use this instead.